rust/connlib/tunnel/src/peer.rs

```diff
@@ -993,155 +1049,49 @@ mod tests {
-    use connlib_shared::messages::gateway::PortRange;
+    use connlib_shared::messages::gateway::{
+        PortRange, ResourceDescription, ResourceDescriptionCidr,
+    };
     use ip_packet::make::{icmp_request_packet, tcp_packet, udp_packet};
     use proptest::{
         arbitrary::any,
         collection, prop_oneof,
         sample::select,
         strategy::{Just, Strategy},
     };
-    use std::ops::RangeInclusive;
+    use std::{collections::BTreeSet, ops::RangeInclusive};
     use test_strategy::Arbitrary;
 
     #[test_strategy::proptest()]
     fn gateway_accepts_allowed_packet(
         #[strategy(client_id())] client_id: ClientId,
-        #[strategy(vec![resource_id(); 5])] resources_id: Vec<ResourceId>,
+        #[strategy(cidr_resources(filters_with_allowed_protocol(), 5))] resources: Vec<(
+            ResourceDescription,
+            Protocol,
+            IpAddr,
+        )>,
         #[strategy(any::<Ipv4Addr>())] src_v4: Ipv4Addr,
         #[strategy(any::<Ipv6Addr>())] src_v6: Ipv6Addr,
-        #[strategy(cidr_with_host())] config: (IpNetwork, IpAddr),
-        #[strategy(collection::vec(filters_with_allowed_protocol(), 1..=5))] protocol_config: Vec<
-            (Filters, Protocol),
-        >,
         #[strategy(any::<u16>())] sport: u16,
         #[strategy(any::<Vec<u8>>())] payload: Vec<u8>,
     ) {
-        let (resource_addr, dest) = config;
-        let src = if dest.is_ipv4() {
-            src_v4.into()
-        } else {
-            src_v6.into()
-        };
-        let mut filters = protocol_config.iter();
         // This test could be extended to test multiple src
         let mut peer = ClientOnGateway::new(client_id, src_v4, src_v6);
-        let mut resource_addr = Some(resource_addr);
-        let mut resources = 0;
-
-        loop {
-            let Some(addr) = resource_addr else {
-                break;
-            };
-            let Some((filter, _)) = filters.next() else {
-                break;
-            };
-            peer.add_resource(
-                vec![addr],
-                resources_id[resources],
-                filter.clone(),
-                None,
-                None,
-            );
-            resources += 1;
-            resource_addr = supernet(addr);
-        }
-
-        for (_, protocol) in &protocol_config[0..resources] {
-            let packet = match protocol {
-                Protocol::Tcp { dport } => tcp_packet(src, dest, sport, *dport, payload.clone()),
-                Protocol::Udp { dport } => udp_packet(src, dest, sport, *dport, payload.clone()),
-                Protocol::Icmp => icmp_request_packet(src, dest, 1, 0, &[]),
-            }
-            .unwrap();
-            assert!(peer.ensure_allowed_dst(&packet).is_ok());
+        for (resource, _, _) in &resources {
+            peer.add_resource(resource.clone(), None);
         }
-    }
-
-    #[test_strategy::proptest()]
-    fn gateway_accepts_allowed_packet_multiple_ips_resource(
```

Comment:
This test didn't make sense to me. A single CIDR resource can only ever have one network. Due to the nature of a proptest, we will try different IPs within networks in the other test already so I think this is redundant?

Reply:
The idea behind this test is that the filters work with multiple either overlapping or non-overlapping range of ips

Reply:
as far as I remember all other tests use a single range of ips

Reply:
I am not sure what you mean. Given that you will be touching this soon, do you mind re-adding a test that you think needs to exist here?

Reply:
yeah, I can do it, but to maybe clarify it, all the other tests have a single ip entry in the `IpNetworkTable` for the `FilterEngine`, this test have multiple of them.

Reply:
It's important because we don't trivially just add the entries to the table but we merge them before adding them.